src/renderer_server/state.rs

```diff
@@ -40,6 +47,32 @@ impl Default for DrawingState {
     }
 }
 
+impl DrawingState {
+    pub(crate) fn to_debug(&self) -> debug::Drawing {
+        let &Self {
```

Comment:
Would it be possible to return a `fmt::Debug` or a string here instead of the whole struct? as that would mean we wouldn't have to duplicate the drawing and turtle state structs

Reply:
There is a comment [here](https://github.com/sunjay/turtle/blob/6ef4a543d62d000ebc2c0d656bd019b6d64f70b8/src/debug.rs#L1-L9) that explains this decision a bit more. To summarize: We can't really avoid duplicating the state because the printable version isn't exactly the same (compare the fields). We also don't really *want* to avoid duplicating it because we want to keep it flexible in case we need to change how we represent the state so it's better for people debugging.

Regarding your specific suggestions: You can't actually prepare a string in advance because the formatting options (set in the `Formatter` struct) aren't available until the `Debug` trait method is actually being run. We also wouldn't really even want to because allocating a string for this is kind of wasteful when the formatting traits usually avoid that allocation.

I definitely appreciate the thought though! In general, duplication isn't always bad. In this case it's actually quite useful! :)